S2VX.Game/Story/Note/SliderPathDrawNode.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Batches;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Graphics.OpenGL;
+using osu.Framework.Graphics.OpenGL.Vertices;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Graphics.Shaders;
+using osu.Framework.Utils;
+using osuTK;
+using osuTK.Graphics;
+using osuTK.Graphics.ES30;
+using System;
+using System.Collections.Generic;
+
+namespace S2VX.Game.Story.Note {
+    public partial class SliderPath {
+        private class SliderPathDrawNode : DrawNode {
+            protected new SliderPath Source => (SliderPath)base.Source;
+
+            private readonly List<Line> Segments = new();
+
+            private Vector2 DrawSize;
+            private float Radius;
+            private IShader PathShader;
+
+            private readonly LinearBatch<TexturedVertex3D> LinearBatch = new(4 * 100 * 3, 10, PrimitiveType.Triangles);
```

Comment:
Are these numbers supposed to mean something? Can we add a comment explaining them

Reply:
I don't really know what these do. I put in osu!'s comment that they had in Path_DrawNode.cs:

```csharp
            // We multiply the size param by 3 such that the amount of vertices is a multiple of the amount of vertices
            // per primitive (triangles in this case). Otherwise overflowing the batch will result in wrong
            // grouping of vertices into primitives.
            private readonly LinearBatch<TexturedVertex3D> halfCircleBatch = new LinearBatch<TexturedVertex3D>(MAX_RES * 100 * 3, 10, PrimitiveType.Triangles);
            private readonly QuadBatch<TexturedVertex3D> quadBatch = new QuadBatch<TexturedVertex3D>(200, 10);
```